### ISSUE TITLE
build scripts: set MACOSX_DEPLOYMENT_TARGET env var for OS X builds

### ISSUE THREAD

**Issue by aerickson** [mpv-player/mpv#5771]
When I compile mpv on Mac OS X 10.11 with the OS X app bundle I get an error when trying to launch the app. I found https://github.com/mpv-player/mpv/commit/c176f1169f356059ee6e3c447e4546fcdb335ec2 and setting it fixes the issue. Setting it to 10.3 does give some compilation warnings, but it works. 10.9 or 10.10 may be more reasonable.

If someone could point me at the proper place in the build files to set this environment variable I'm happy to make a pull request.

Thanks.

related: https://github.com/Homebrew/homebrew-core/pull/27127

### mpv version and platform

Latest master, running on Mac OS X 10.11.6.

```
mpv 65f0825 Copyright © 2000-2018 mpv/MPlayer/mplayer2 projects
 built on Thu Apr 26 04:43:24 EDT 2018
ffmpeg library versions:
   libavutil       56.15.100
   libavcodec      58.19.100
   libavformat     58.13.100
   libswscale      5.2.100
   libavfilter     7.19.100
   libswresample   3.2.100
ffmpeg version: git-2018-04-22-153e920
```

### Reproduction steps

```
// be on a OS X 10.11 system
./bootstrap.py
python3 waf configure --enable-libmpv-shared
python3 waf build
python3 TOOLS/osxbundle.py build/mpv
// cd build
// double click to launch mpv.app
```

### Expected behavior

The app starts.

### Actual behavior

The app does not start and gives and OS X gives the following error:

```
You can’t use this version of the application “mpv” with this version of OS X.

You have OS X 10.11.6. The application requires OS X 10.12 or later.
```

### Log file

not relevant

### Sample files

not relevant

**Comment by Akemi** [comment]
as far as I am concerned this is a 10.11 (SDK)/swift bug that was introduced with the cocoa-cb backend and swift support, one of many that were fixed with the a newer SDK/swift. the build tools are supposed to pick the current running macOS version as a deployment target when nothing is explicitly set, which it doesn't on 10.11. it makes no sense that they pick 10.12 on 10.11.

I tested it on 10.12 and 10.13 with different SDKs and there it works without any problems. so apple most likely considers this bug fixe and doesn't care for 10.11 anymore. i could even reproduce this bug with a minimal example outside of mpv, on 10.11 only.

hardcoding a deployment target is bad practice and shouldn't be done. picking 10.3 is also kinda bad since the latest version i support is 10.10. on a side note using `--disable-macos-cocoa-cb` should fix that problem too, since it deactivates all the swift parts completely.

when 10.14 is released i will probably drop support for 10.11 for the cocoa-cb backend because of the many bugs i encountered there.

there is small possibility that there is a mistake somewhere in the swift build parts i wrote. but as long as this isn't proven i don't consider this our bug.

**Comment by Argon-** [comment]
Fwiw, I've had builds on 10.11 that did work and didn't have this mentioned problem here. Only a tiny fraction among all of the `You can’t use this version of the application “mpv” with this version of OS X.` builds but some did work. I never figured out the reason for why it worked sometimes. Probably because I didn't really invest time into that, as I rarely use the bundle and often only noticed it by accident in a "huh, why does this work" fashion.
It really felt random and nondeterministic.
And now that I'm on 10.13 I can't really test it anymore.
